skyportal/onboarding.py

```diff
@@ -110,19 +120,45 @@ def setup_invited_user_permissions(strategy, uid, details, user, *args, **kwargs
     group_ids = [g.id for g in invitation.groups]
 
     existing_user_group_ids = [g.id for g in existing_user.groups]
-    # Add user to specified groups
-    for group_id, admin in zip(group_ids, invitation.admin_for_groups):
-        if group_id not in existing_user_group_ids:
-            DBSession.add(GroupUser(user_id=user.id, group_id=group_id, admin=admin))
+
+    # grab stream_ids
+    stream_ids = set()
+    for group_id in group_ids:
+        streams = (
+            DBSession()
+            .query(Stream)
+            .join(GroupStream)
+            .filter(GroupStream.group_id == group_id)
+            .all()
+        )
+        for stream in streams:
+            stream_ids.add(stream.id)
 
     if DBSession().query(Group).filter(Group.name == user.username).first() is None:
         # Create single-user group
         DBSession().add(Group(name=user.username, users=[user], single_user_group=True))
+        # Add stream access to single user group
+        single_user_group = (
+            DBSession().query(Group).filter(Group.name == user.username).first()
+        )
+        for stream_id in stream_ids:
+            DBSession.add(
+                GroupStream(group_id=single_user_group.id, stream_id=stream_id)
+            )
+            DBSession.add(StreamUser(stream_id=stream_id, user_id=user.id))
```

Comment:
Should this be outside of this if block? We want to do this even if the single user group might exist, right?

Reply:
Does this even need a block (i.e., do we need the `if` on 137)? How would the single user group exist already for an invited user?

Reply:
yeah, that can probably be removed. It was an extra precaution during the development phase, but I think the checks prior to this point are robust enough

Reply:
Ok, removed the check